docs/nf4_science/rnaseq/03_multi-sample.md

```diff
@@ -0,0 +1,489 @@
+ N E X T F L O W   ~  version 24.10.0
+
+Launching `rnaseq.nf` [golden_curry] DSL2 - revision: 2a5ba5be1e
+
+executor >  local (18)
+[07/3ff9c5] FASTQC (6)       [100%] 6 of 6 ✔
+[cc/16859f] TRIM_GALORE (6)  [100%] 6 of 6 ✔
+[68/4c27b5] HISAT2_ALIGN (6) [100%] 6 of 6 ✔
+```
+
+That's all it took to get the workflow to run on multiple files!
+Nextflow handles all the parallelism for us.
+
+---
+
+## 2. Aggregate pre-processing QC metrics into a single MultiQC report
+
+All this produces a lot of QC reports, and we don't want to have to dig through individual reports.
+This is the perfect point to put in a MultiQC report aggregation step!
+
+### 2.1. Create a module for the QC aggregation process
+
+Let's create a module file called `modules/multiqc.nf` to house the `MULTIQC` process:
+
+```bash
+touch modules/multiqc.nf
+```
+
+Open the file in the code editor and copy the following code into it:
+
+```groovy title="modules/multiqc.nf" linenums="1"
+#!/usr/bin/env nextflow
+
+process MULTIQC {
+
+    container "community.wave.seqera.io/library/pip_multiqc:ad8f247edb55897c"
+    publishDir "results/multiqc", mode: 'copy'
+
+    input:
+    path '*'
+    val output_name
+
+    output:
+    path "${output_name}.html", emit: report
+    path "${output_name}_data", emit: data
+
+    script:
+    """
+    multiqc . -n ${output_name}.html
+    """
+}
+```
+
+### 2.2. Import the module into the workflow file
+
+Add the statement `include { MULTIQC } from './modules/multiqc.nf'` to the `rnaseq.nf` file:
+
+```groovy title="rnaseq.nf" linenums="3"
+// Module INCLUDE statements
+include { FASTQC } from './modules/fastqc.nf'
+include { TRIM_GALORE } from './modules/trim_galore.nf'
+include { HISAT2_ALIGN } from './modules/hisat2_align.nf'
+include { MULTIQC } from './modules/multiqc.nf'
+```
+
+### 2.3. Add a `report_id` parameter and give it a sensible default
+
+```groovy title="rnaseq.nf" linenums="9"
+/*
+ * Pipeline parameters
+ */
+params.hisat2_index_zip = "data/genome_index.tar.gz"
+params.report_id = "all_single-end"
+```
+
+### 2.4. Call the process on the outputs of the previous steps
+
+We need to give the `MULTIQC` process all the QC-related outputs from previous steps.
+
+For that, we're going to use the `.mix()` operator, which aggregates multiple channels into a single one.
+
+If we had four processes called A, B, C and D with a simple `.out` channel each, the syntax would look like this: `A.out.mix( B.out, C.out, D.out )`. As you can see, you apply it to the first of the channels you want to combine (doesn't matter which) and just add all the others, separated by commas, in the parenthesis that follows.
+
+In the case of our workflow, we have the following outputs to aggregate:
+
+- `FASTQC.out.zip`
+- `FASTQC.out.html`
+- `TRIM_GALORE.out.trimming_reports`
+- `TRIM_GALORE.out.fastqc_reports`
+- `HISAT2_ALIGN.out.log`
+
+So the syntax example becomes:
+
+```groovy title="Applying .mix() in the MULTIQC call"
+        FASTQC.out.zip.mix(
```

Comment:
```suggestion
        Channel.empty.mix(
            FASTQC.out.zip,
```

Reply:
TIL

Reply:
Same as below, going to keep that for the next pass of improvements. Great idea, just can't deal with domino updates right now.